### ISSUE TITLE
Company candidate number disappear

### ISSUE THREAD

**Issue by azzamsa** [sebastiencs/company-box#54]
Hi.

It's working fine `(setq company-show-numbers t)` without company-box. Can I get company candidate number back while using company-box ?

Thank you for company-box.

**Comment by testinggithub1222** [comment]
I think so. I miss those numbers and figure out that company-box make it disappear. so sad both are really helpful and now we have to choose one over the other.

**Comment by azzamsa** [comment]
I choose being fast rather than shiny. I use 'vanilla' company for that, it's hard to have completion without being able to do it fast (have to use arrow and so on). I prefer to use number and drop this.

**Comment by testinggithub1222** [comment]
@azzamsa even without showing the number, i still can type using M-number but only if you willing to count it while typing.

**Comment by azzamsa** [comment]
I prefer not to. :)

**Comment by testinggithub1222** [comment]
:)

**Comment by whirm** [comment]
I'd like to have this reopened, it's quite annoying to not have numbers.

**Comment by azzamsa** [comment]
I closed this cause there is no response from it's maintainer over long period of time.

**Comment by CsBigDataHub** [comment]
@whirm @azzamsa 
Here is a lazy fix I made, essentially over-riding the `company-box--render-buffer` function to display line numbers. Add this to your configuration after `company-box`. This will show line numbers for the company buffer.

```
(defun company-box--render-buffer (string)
  (let ((selection company-selection))
    (with-current-buffer (company-box--get-buffer)
      (erase-buffer)
      (insert string "\n")
      (setq mode-line-format nil
            display-line-numbers t
            truncate-lines t
            cursor-in-non-selected-windows nil)
      (setq-local scroll-step 1)
      (setq-local scroll-conservatively 10000)
      (setq-local scroll-margin  0)
      (setq-local scroll-preserve-screen-position t)
      (add-hook 'window-configuration-change-hook 'company-box--prevent-changes t t)
      (company-box--update-line selection))))
```

![image](https://user-images.githubusercontent.com/23665525/74557676-43056500-4f26-11ea-9bb4-e28aca912932.png)

@sebastiencs Are there any plans to enable `company-show-numbers` in `company-box` ?

**Comment by lsiksous** [comment]
Thx @CsBigDataHub you're right. That should be customizable. I added (setq header-line-format nil) too.

**Comment by sebastiencs** [comment]
`company-show-numbers` is now supported in `company-box`:

![image](https://user-images.githubusercontent.com/5273820/89728358-5e79a480-da5f-11ea-84e0-dd1514823696.png)

**Comment by tshu-w** [comment]
@sebastiencs If it's possible to add margin for company number
<img width="559" alt="Screen Shot 2020-08-09 at 11 33 27 PM" src="https://user-images.githubusercontent.com/13161779/89735974-bc76ae00-da98-11ea-8a35-ff51c36a6b13.png">

**Comment by azzamsa** [comment]
confirmed. solved using `981aa41` revision.

Thanks @sebastiencs

**Comment by sebastiencs** [comment]
@tshu-w Done in https://github.com/sebastiencs/company-box/commit/439951e60c5bb9c50245a5a4b3a8cb7122cb6106
